presto-hive/src/main/java/com/facebook/presto/hive/HiveWriteUtils.java

```diff
@@ -376,8 +377,13 @@ public static void renameDirectory(HdfsEnvironment hdfsEnvironment, String schem
     public static Path createTemporaryPath(HdfsEnvironment hdfsEnvironment, Path targetPath)
     {
         // use a per-user temporary directory to avoid permission problems
-        // TODO: this should use Hadoop UserGroupInformation
-        String temporaryPrefix = "/tmp/presto-" + StandardSystemProperty.USER_NAME.value();
+        String temporaryPrefix;
+        try {
+            temporaryPrefix = "/tmp/presto-" + UserGroupInformation.getCurrentUser().getUserName();
```

Comment:
Wouldn't this tmp dir need to be in the users home dir, or somewhere where they have write permissions?